### ISSUE TITLE
"Use local function" introduces CS1503 when passed to an overload that takes both a delegate & value

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#26526]
**Steps to Reproduce**:

Couldn't break it down any further without Moq package installed:

``` C#
using System;
using Moq;

class Program
{
    static void Main(string[] args)
    {
        Func<IProjectTree, string> getPath = tree => tree.FilePath; <!-- Use local function

        var mock = new Mock<IProjectTreeProvider>();
        mock.Setup(p => p.GetPath(It.IsAny<IProjectTree>()))
            .Returns(getPath);

    }
}

internal interface IProjectTreeProvider
{
    string GetPath(IProjectTree projectTree);
}

internal interface IProjectTree
{
    string FilePath { get; set; }
}
```
**Expected Behavior**:

``` C#
    static void Main(string[] args)
    {
        string getPath(IProjectTree tree) => tree.FilePath;

        var mock = new Mock<IProjectTreeProvider>();
        mock.Setup(p => p.GetPath(It.IsAny<IProjectTree>()))
            .Returns((Func<IProjectTree, string>)getPath);

    }
```

**Actual Behavior**:

``` C#
    static void Main(string[] args)
    {
        string getPath(IProjectTree tree) => tree.FilePath;

        var mock = new Mock<IProjectTreeProvider>();
        mock.Setup(p => p.GetPath(It.IsAny<IProjectTree>()))
            .Returns(getPath);

    }

```

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1503	Argument 1: cannot convert from 'method group' to 'string'	ConsoleApp299	C:\Users\davkean\Source\Repos\ConsoleApp299\ConsoleApp299\Program.cs	12	Active

```

**Comment by Neme12** [comment]
Simpler repro:
```c#
using System;

class C
{
    void M()
    {
        Func<string> f = () => "";

        var f2 = f;
    }
}
```
```c#
using System;

class C
{
    void M()
    {
        string f() => "";

        var f2 = f; // error CS0815: Cannot assign method group to an implicitly-typed variable
    }
}
```

**Comment by Neme12** [comment]
This should be equivalent to the original repro:
```c#
using System;

class C
{
    void M()
    {
        Func<int, string> f = o => "";

        Method(f);
    }

    void Method<T>(Func<T, string> o)
    {
    }

    void Method(string o)
    {
    }
}
```
The second overload is not actually needed. What fails is generic type inference.

**Comment by Neme12** [comment]
I'd like to fix this but would prefer to wait until #24763 gets resolved which fixes tons of other bugs inside UseLocalFunction, as opposed to giving myself a lot merge conflicts and possibly duplicating some work.
